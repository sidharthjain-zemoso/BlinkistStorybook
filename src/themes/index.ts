import { createTheme } from "@mui/material/styles";

declare module "@mui/material/Typography" {
  interface TypographyPropsVariantOverrides {
    heading: true;
    heading3: true;
    subtitle3: true;
    body3: true;
    caption1: true;
    caption2: true;
    caption3: true;
  }
}

declare module "@mui/material/styles" {
  interface TypographyVariants {
    heading: React.CSSProperties;
    heading3: React.CSSProperties;
    subtitle3: React.CSSProperties;
    body3: React.CSSProperties;
    caption1: React.CSSProperties;
    caption2: React.CSSProperties;
    caption3: React.CSSProperties;
  }

  interface TypographyVariantsOptions {
    heading?: React.CSSProperties;
    heading3: React.CSSProperties;
    subtitle3?: React.CSSProperties;
    body3?: React.CSSProperties;
    caption1?: React.CSSProperties;
    caption2?: React.CSSProperties;
    caption3: React.CSSProperties;
  }

  interface PaletteColor {
    darker?: string;
    border?: string;
    placeholder?: string;
  }

  interface SimplePaletteColorOptions {
    darker?: string;
    placeholder?: string;
    border?: string;
  }

  interface Palette {
    greyscale: Palette["primary"];
    textcolor: Palette["primary"];
    backgroundcolor: Palette["primary"];
  }

  interface PaletteOptions {
    greyscale?: PaletteOptions["primary"];
    textcolor?: PaletteOptions["primary"];
    backgroundcolor?: PaletteOptions["primary"];
  }
}

export const theme = createTheme({
  components: {
    MuiButtonBase: {
      styleOverrides: {
        root: {
          // "&:hover": {
          //   backgroundColor: "none",
          // },
        },
      },
    },
  },
  spacing: 4,
  palette: {
    primary: {
      light: "#E2F2E9",
      main: "#2CE080",
      dark: "#22C870",
      darker: "#20BA68",
    },
    secondary: {
      light: "#DFE8F6",
      main: "#0365F2",
    },
    greyscale: {
      light: "#BAC9CF",
      main: "#6C787F",
      dark: "#3A4649",
      darker: "#042330",
    },
    textcolor: {
      light: "#6D787E",
      main: "#03314B",
      placeholder: "#747575",
      dark: "#042330",
    },
    backgroundcolor: {
      light: "#F7F7F7",
      main: "#F1F6F4",
    },
  },
  typography: {
    heading: {
      fontSize: "36px",
      lineHeight: "45px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    heading3: {
      fontSize: "24px",
      lineHeight: "30.17px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    subtitle1: {
      fontSize: "24px",
      lineHeight: "32px",
      // fontWeight: 500,
      fontFamily: "Cera Pro Medium",
    },
    subtitle2: {
      fontSize: "18px",
      lineHeight: "24px",
      fontWeight: 700,
      fontFamily: "Cera Pro Light",
    },
    subtitle3: {
      fontSize: "18px",
      lineHeight: "24px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    body1: {
      fontSize: "16px",
      lineHeight: "20px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    body2: {
      fontSize: "16px",
      lineHeight: "24px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    body3: {
      fontSize: "22px",
      lineHeight: "20px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    caption1: {
      fontSize: "14px",
      lineHeight: "22px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    caption2: {
      fontSize: "14px",
      lineHeight: "22px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    caption3: {
      fontSize: "14px",
      lineHeight: "17.6px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
  },
});

// theme.overrides = {
//   ...theme.overrides,
//   MuiButton: {
//     root: {
//     }
//   }
// }
