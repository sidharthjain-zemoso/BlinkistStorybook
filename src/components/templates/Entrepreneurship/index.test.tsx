import { render, screen } from "@testing-library/react";
import EntrepreneurshipTemplate from "./";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@mui/styles";
import { theme } from "../../../themes";

const DummyView = () => <div>Dummy View</div>;

const booksInfo = [
  {
    id: 0,
    isFinishedReading: true,
    isAddedToLib: true,
    category: "TrendingBlinks",
    title: "Bring Your Human to Work",
    author: "Erica Keswin",
    time: "13",
    reads: "",
    imageSrc: "1",
  },
];

const MockTemplate = () => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <EntrepreneurshipTemplate
          header={<DummyView />}
          footer={<DummyView />}
          searchBar={<DummyView />}
          cardBanner={<DummyView />}
          blinks={booksInfo}
        />
      </ThemeProvider>
    </BrowserRouter>
  );
};

describe("MyLibrary Template", () => {
  it("Render MyLibrary Template", () => {
    render(<MockTemplate />);
    const template = screen.getByTestId("entrepreneurship-template-id");
    expect(template).toBeDefined();
    expect(template).toMatchSnapshot();
  });
});
