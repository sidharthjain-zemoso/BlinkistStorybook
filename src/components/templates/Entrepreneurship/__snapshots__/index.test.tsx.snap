// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLibrary Template Render MyLibrary Template 1`] = `
<div
  class="entrepreneurship-container"
  data-testid="entrepreneurship-template-id"
>
  <div>
    Dummy View
  </div>
  <div
    class="MuiBox-root css-but0jo"
  >
    <div
      class="MuiBox-root css-7822g1"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10 MuiGrid-direction-xs-column css-wldefx-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-pqkznn-MuiGrid-root"
        >
          <div>
            Dummy View
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-1kfl3yl-MuiGrid-root"
        >
          <div>
            Dummy View
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-heading3 css-37jdci-MuiTypography-root"
            style="margin-bottom: 15px;"
          >
            Trending Blinks
          </div>
          <div
            class="MuiBox-root css-tcwb5a"
            data-testid="trending-blinks-id"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-bookCardStyles-1 makeStyles-cardContainer-2 css-31xq66-MuiPaper-root-MuiCard-root"
              data-testid="book-card-id"
            >
              <img
                alt="Book Cover"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="350"
                src="/res/cover images/1.png"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="card-content-wrapper"
                >
                  <div
                    class="makeStyles-cardContentWrapperELement-3"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-subtitle1 css-14vjfu0-MuiTypography-root"
                    >
                      Bring Your Human to Work
                    </div>
                  </div>
                  <div
                    class="makeStyles-cardContentWrapperELement-3"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-body1 css-1m75f5p-MuiTypography-root"
                    >
                      Erica Keswin
                    </div>
                  </div>
                  <div
                    class="makeStyles-readElement-4 makeStyles-cardContentWrapperELement-3"
                  >
                    <div
                      class="read-time"
                      style="display: flex; align-items: center;"
                    >
                      <img
                        alt="icon"
                        src="/res/icons/clock.png"
                        style="margin-right: 5px;"
                      />
                      <div
                        class="MuiTypography-root MuiTypography-caption3 css-1gmjhmd-MuiTypography-root"
                      >
                        13
                        -minute read
                      </div>
                    </div>
                    
                  </div>
                  
                </div>
              </div>
              <button
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root makeStyles-addToLib-5 css-r1tzy3-MuiButtonBase-root-MuiButton-root"
                id="cardButtonId"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AddIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                </span>
                <div
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Add To Library
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-heading3 css-37jdci-MuiTypography-root"
            style="margin-bottom: 15px;"
          >
            Just added
          </div>
          <div
            class="MuiBox-root css-tcwb5a"
            data-testid="just-added-blinks-id"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-heading3 css-37jdci-MuiTypography-root"
            style="margin-bottom: 15px;"
          >
            Featured audio Blinks
          </div>
          <div
            class="MuiBox-root css-tcwb5a"
            data-testid="featured-audio-blinks-id"
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    Dummy View
  </div>
</div>
`;
